### ISSUE TITLE
plans for laravel 5?

### ISSUE THREAD

**Issue by Goddard** [JVMartin/angel-core#4]
Just curious if this is still active and if you are looking toward laravel 5?

**Comment by JVMartin** [comment]
Hey @Goddard - thanks for the question, yes indeed!

I'm a full-time computer science student now so I don't have as much time to work on Angel lately, but I have indeed aleady started to rewrite the entire system from scratch in Laravel 5.  It's going to be a lot simpler at first, without the eCommerce plugin and whatnot.  I'm going to apply some better coding principles to the page system and menu system.

It should also be a lot easier to install and customize.  Updates soon!  :)

**Comment by Goddard** [comment]
re-write the whole thing?  I am not sure you would even need to do that.  Your system is pretty nice already.  I think an interface redesign would be better and more marketing.  It could be one of the better Laravel CMS systems out right now.  OcotoberCMS is the only other CMS I like except yours and it has less features then yours, but just looks prettier.

**Comment by JVMartin** [comment]
Thanks man.  I'd definitely like to make the interface prettier!  As far as the code goes, I want it to stay really simple and intuitive...  for instance, the thing I most like about Angel right now is that if you open up the "src" folder, it is laid out just exactly like a Laravel 4 "app" folder with:
- controllers
- models
- filters.php
- routes.php

I'd like to do the same thing with Laravel 5, so I'm shifting everything around to match it, such as:
- Http
- Http\Controllers
- Http\Middleware
- Events
- Services

A lot of things that used to go right in models can now become services and events in order to leverage the new service-oriented class paradigm.

In addition:  Laravel 5 finally comes with a user registration system, login/signup/password reset pages, etc.  I'm leveraging all of that to replace the systems I had built custom for the old Angel.  It also comes with some really interesting RESTful controller code that might make Angel's CrudController obsolete!

The goal is to have anyone who is already real familiar with Laravel 5 feel right at home in the new Angel system, and to do that I have to refactor the framework the Laravel 5 way.  95% of what makes Angel tick is in the CrudController, so it'd be really cool to make that a _lot_ simpler in the process if I can!

I really appreciate your input on this, please do let me know if you have any advice, I try to make everyone happy with it.  In the end, I hope to not have to rewrite too much of the code, just to kinda refactor it into the new paradigm as much as possible... so that I can get it shipped ASAP.  So I didn't quite mean "from scratch", but it sure feels that way with all of these new changes.

I've been watching October off and on for a while... I think it's really cool, but I don't like that it's a replacement for Laravel instead of just an add-on for it.  By that I mean it's baked right into Laravel whereas Angel is a package that can update separately from Laravel.  It's really sexy for sure, though.  And I may actually end up going the same route and building Angel right into the framework instead of keeping it as a separate package... I'm not sure yet.

Thanks again.

**Comment by Goddard** [comment]
Yeah I hate that about October as well.  They are really trying to make people dependent on them rather then adding to laravel.  The route you are taking is good as long as you are able to handle staying with laravel development.  How many people are helping you with this project?

**Comment by JVMartin** [comment]
As far as writing code goes, Angel has been a solo venture so far.  However, there are a good handful of great developers who use the framework every day and let me know about bugs and whatnot.  I'd love to collaborate with others on it, especially because I'm not very into front-end stuff, and I know it could be a lot prettier.  I've never used Gulp before; so that whole workflow is very new to me with Elixir on Laravel 5.  In the past, I've only done super basic SASS compilation for Foundation and whatnot.  I'm hoping to integrate a cleaner and more robust front-end workflow into Angel using Elixir.

**Comment by vitorf7** [comment]
Hi @JVMartin have you had a look at AdminLTE for the Admin design?

**Comment by JVMartin** [comment]
Hey @vitor-faiante - I hadn't seen that before, it looks really sweet!  Totally thinking about integrating it into the Laravel 5 version of the CMS when we build it out more.
